### DISCUSSION TITLE
How to handle partial data fetching?

### DISCUSSION THREAD

**Discussion by vedantroy** [remix-run/remix#2825] [Q&A]
Imagine the following scenario:
1. I have a page that has a vote count, and a bunch of other data
2. The user clicks the upvote button
3. Remix will update the data using the action function & then call the loader to refetch all the data

The problem here is, it seems very unnecessary to fetch all the other data again (for example, the post text), since the only new value we need is the new vote count.

In-fact, even fetching a new vote count might be bad b/c it's possible that someone else may have also incremented the vote in the meantime & we would not want the vote count to jump up by 2 after the user clicks the upvote button.

What is the recommended practice here?

I think this is somewhat of a premature optimization since I doubt it will cause too much extra load to just refresh the entire page, but I'm curious to know what the recommended approach is.

**Comment by AYastrebtsov** [comment]
I think this is the way to go
https://www.epicweb.dev/full-stack-components
